### PR TITLE
Enable linting support for source code files and fix current lint errors

### PR DESCRIPTION
**Pull request by wak-google**

### CODE REVIEW

**Comment by wak-google** [comment]
An example of linter error output: https://travis-ci.org/wak-google/pblog/jobs/184605508

File: src/record.c [hunk 1]
```diff
@@ -144,16 +144,20 @@ static int log_read_record(struct record_intf *ri, int offset, int *next_offset,
     }
     offset -= region->used_size;
   }
+  if (region == NULL) {
+	return PBLOG_ERR_INVALID;
```
Comment (rlippert) [inline src/record.c]: 2 space indent

File: src/record.c [hunk 2]
```diff
@@ -144,16 +144,20 @@ static int log_read_record(struct record_intf *ri, int offset, int *next_offset,
...
+  }
   if (i >= meta->used_regions) {
     // Check for end of log (reading last record one past end).
     // Return success in that case, set next_offset to 0.
     if (offset == 0 || offset == region->used_size) {
       *next_offset = 0;
-      if (len) *len = 0;
+      if (len) {
+		*len = 0;
```
Comment (rlippert) [inline src/record.c]: this looks off

File: src/record.c (hunk 1 above)
Comment (wak-google) [reply src/record.c]: Fixed

File: src/record.c (hunk 2 above)
Comment (wak-google) [reply src/record.c]: Fixed

File: test/pblog_test.cc [hunk 3]
```diff
@@ -56,32 +56,34 @@ class PblogFileTest : public ::testing::Test {
     file_regions[1].size = size1;
     file_regions[1].used_size = 0;
 
-    flash_ri_ = reinterpret_cast<struct record_intf *>(
+    flash_ri_ = static_cast<struct record_intf *>(
         malloc(sizeof(struct record_intf)));
-    pblog_file_ops.priv = (void *)filename_.c_str();
-    record_intf_init(flash_ri_, file_regions, 2, &pblog_file_ops);
+    pblog_file_ops.priv = static_cast<void *>(&filename_[0]);
```
Comment (rlippert) [inline test/pblog_test.cc]: is &filename_[0] guaranteed to be a null-terminated C string?

File: test/record_test.cc [hunk 4]
```diff
@@ -140,28 +140,27 @@ TEST_F(RecordFileTest, FirstTimeInit) {
...
 
 TEST_F(RecordFileTest, InitWithGarbage) {
   // Fill the file with garbage.
-  string garbage(4096, 0);
+  string garbage(4096, '\0');
   for (size_t i = 0; i < garbage.size(); ++i) {
     garbage[i] = i;
   }
-  FILE *file = fopen(filename_.c_str(), "w");
+  std::unique_ptr<FILE, int(*)(FILE *)> file(fopen(filename_.c_str(), "w"), fclose);
```
Comment (rlippert) [inline test/record_test.cc]: should put this in a separate scope { } to guarantee that the file is closed/flushed before reading

File: test/pblog_test.cc (hunk 3 above)
Comment (wak-google) [reply test/pblog_test.cc]: Done

File: test/record_test.cc (hunk 4 above)
Comment (wak-google) [reply test/record_test.cc]: Yes

File: test/pblog_test.cc (hunk 3 above)
Comment (wak-google) [reply test/pblog_test.cc]: No it has no guarantees but libstdc++ has it this way :) I figured out a way for the linter to not be annoying about the const cast.

**Review by rlippert — APPROVED** [review]

**Pull request merged**